crates/sui-sdk/src/wallet_context.rs

```diff
@@ -346,4 +347,104 @@ impl WalletContext {
+        get_new_package_obj_from_response(&resp).unwrap()
+    }
+
+    /// Executes a transaction to publish the `nfts` package and returns the package id, id of the gas object used, and the digest of the transaction.
+    pub async fn publish_nfts_package(&self) -> (ObjectID, ObjectID, TransactionDigest) {
+        let accounts_and_objs = self.get_all_accounts_and_gas_objects().await.unwrap();
+        let sender = accounts_and_objs[0].0;
+        let gas_object = accounts_and_objs[0].1[0];
+        let gas_id = gas_object.0;
+        let gas_price = self.get_reference_gas_price().await.unwrap();
+        let txn = self.sign_transaction(
+            &TestTransactionBuilder::new(sender, gas_object, gas_price)
+                .publish_examples("nfts")
+                .build(),
+        );
+        let resp = self.execute_transaction_block(txn).await.unwrap();
+        let package_id = get_new_package_obj_from_response(&resp).unwrap().0;
+        (package_id, gas_id, resp.digest)
+    }
+
+    /// Pre-requisite: `publish_nfts_package` must be called before this function.
+    /// Executes a transaction to create an NFT and returns the sender address, the object id of the NFT, and the digest of the transaction.
+    pub async fn create_devnet_nft(
+        &self,
+        package_id: ObjectID,
+    ) -> (SuiAddress, ObjectID, TransactionDigest) {
+        let accounts_and_objs = self.get_all_accounts_and_gas_objects().await.unwrap();
+        let sender = accounts_and_objs[0].0;
+        let gas_object = accounts_and_objs[0].1[0];
+        let rgp = self.get_reference_gas_price().await.unwrap();
+
+        let txn = self.sign_transaction(
+            &TestTransactionBuilder::new(sender, gas_object, rgp)
+                .call_nft_create(package_id)
+                .build(),
+        );
+        let resp = self.execute_transaction_block(txn).await.unwrap();
+
+        let object_id = resp
+            .effects
+            .as_ref()
+            .unwrap()
+            .created()
+            .first()
+            .unwrap()
+            .reference
+            .object_id;
+
+        (sender, object_id, resp.digest)
+    }
+
+    /// Executes a transaction to delete the given NFT.
+    pub async fn delete_devnet_nft(
+        &mut self,
+        sender: SuiAddress,
+        package_id: ObjectID,
+        nft_to_delete: ObjectRef,
+    ) -> SuiTransactionBlockResponse {
+        let gas = self
+            .get_one_gas_object_owned_by_address(sender)
+            .await
+            .unwrap()
+            .unwrap_or_else(|| panic!("Expect {sender} to have at least one gas object"));
+        let rgp = self.get_reference_gas_price().await.unwrap();
+        let txn = self.sign_transaction(
+            &TestTransactionBuilder::new(sender, gas, rgp)
+                .call_nft_delete(package_id, nft_to_delete)
+                .build(),
+        );
+        self.execute_transaction_block(txn).await.unwrap()
+    }
```

Comment:
All the nft functions are highly customized in terms of what they do and return. I think it's better to move them to a separate file called something like devet_nft_transaction_utils